lib/src/postgresql_impl/connection.dart

```diff
@@ -10,7 +10,7 @@ class ConnectionImpl implements Connection {
       TypeConverter typeConverter,
       String getDebugName())
     : _userName = settings.user,
-      _passwordHash = _md5s(settings.password + settings.user),
+      _passwordHash = _md5s(Uri.decodeFull(settings.password) + settings.user),
```

Comment:
Thanks for the input. I've moved the encoding/decoding to the settings class as you suggested.